### ISSUE TITLE
Issue when using more than one loader

### ISSUE THREAD

**Issue by mamantoha** [crystal-i18n/i18n#3]
Hi @ellmetha 

This issue is easily reproducible.

```
require "i18n"

I18n.config.loaders << I18n::Loader::YAML.new("#{__DIR__}/locales")
I18n.config.loaders << I18n::Loader::YAML.new("#{__DIR__}/new_locales")
I18n.config.default_locale = "en"

I18n.init

puts I18n.t("message")
puts I18n.t("new_message")
```

```
en:
  message: "Message"
```

```
en:
  new_message: "New Message"
```

Output:

```
missing translation: en.message
New Message
```

The problem is in this line: https://github.com/crystal-i18n/i18n/blob/732fbc9194752cfd26cc47ddbf395c760770dc31/src/i18n/config.cr#L138

The solution is to use `deep_merge` method from https://github.com/peterhoeg/deep-merge.cr or implement something similar inside this shard.

What do you think?

**Comment by ellmetha** [comment]
Hey, thanks a lot for reporting this! This should be fixed by 8d6f29eb49178d09e7dd64697ef7e8877435c57f (I ended up using a similar implementation inside the shard itself). 😉 

Version `0.1.4` was released with the fix too.

**Comment by mamantoha** [comment]
Thanks for the quick fix.
Btw, yesterday I ended up with this code with the same result:

```
def deep_merge_translations_hash(target, other_hash)
  target.merge!(other_hash) do |key, this_val, other_val|
    if this_val.is_a?(Hash) && other_val.is_a?(Hash)
      deep_merge_translations_hash(this_val, other_val)
    else
      other_val
    end
  end
end
```
